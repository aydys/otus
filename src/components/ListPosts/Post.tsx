import React from "react";

type Props = {
  post: {},
  onClick: (id: number) => void
}

export class Post extends React.Component<Props> {
  render() {
    return <div>Post</div>
  }
}