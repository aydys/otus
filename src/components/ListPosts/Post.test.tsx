import React from "react";
import { shallow } from "enzyme";
import { Post } from "./Post";

describe('Post', () => {

  const props = {
    post: {
      title: '',
      id: 0,
      body: ''
    },
    onClick: jest.fn()
  }

 
  it('layout dom elements (snapshot)', () => {
    const wrapper = shallow(<Post {...props} />)

    expect(wrapper).toMatchSnapshot()
  })

  describe('Post with props', () => {
    const nextProps = {
      ...props,
      post: {
        ...props.post,
        title: 'Some title'
      }
    }

    const wrapper = shallow(<Post {...nextProps} />)

    it('render Title of Post', () => {
      expect(wrapper.find('h4').text()).toEqual(nextProps.post.title);
    })

    it('simulate click to button', () => {
      wrapper.find('button').simulate('click', {
        preventDefault: () => {}
      })

      expect(nextProps.onClick).toHaveBeenCalledTimes(1)
    })
  })  
})
