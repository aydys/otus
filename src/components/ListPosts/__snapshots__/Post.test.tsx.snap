// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post layout dom elements (snapshot) 1`] = `ShallowWrapper {}`;
